src/store/normalize-data.js

```diff
@@ -256,7 +326,8 @@ const normalizeData = (data, expandAllPipelines) => {
     data.layers.forEach(addLayer(state));
   }
 
-  return state;
+  const updatedState = getNodeFiltersFromUrl(state, data.tags);
+  return { ...state, ...updatedState };
```

Comment:
is there a way where we just return state as before and have modifications to the state done within, instead of patching like this ? Like making getNodeFiltersFromUrl function (does not actually getNodeFilters but updates state with filters) as updateStateWithFilters which internally modifies the state. Also data.tags is a bit confusing. Can we do -

updateStateWithFilters(state, data.filters) - which takes care of modifying the state and data.filters contains both tags and nodeType info ? After this function is called we just return state (in the current implementation we are anyway returning - `return { ...state, ...updatedStateWithTags, ...updatedStateWithTypes };`

Reply:
@ravi-kumar-pilla modified and renamed to updateStateWithFilters  and removed patching. here `data.tags` is pass just for tag where `data.tags` contains all the tags.